source/language/directives.rst

```diff
@@ -1,14 +1,95 @@
 Directives
 ==========
 
-OpenQASM supports a directive mechanism that allows other information to
-be included in the program. A directive begins with ``#pragma`` and
-terminates at the end of the line. Directives can provide annotations
-that give additional information to compiler passes and the target
-system or simulator. Ideally the meaning of the program does not change
+OpenQASM supports directive mechanisms that allows other information to
+be included in the program. Directives are either pragmas or annotations.
+Both are used to supply additional information to the compiler passes and the
+target system or simulator. Ideally the meaning of the program does not change
 if some or all of the directives are ignored, so they can be interpreted
 at the discretion of the consuming process.
 
+Pragmas
+-------
+
+Pragma directives start with ``pragma`` and continue to the end of line. The
+text after ``pragma`` is a single string, and parsing is left to the specific
+implementation. The ``#pragma`` syntax is deprecated, but is accepted for
+compatibility. Pragmas should be processed as soon as they are encountered; if a
```

Comment:
Should we maintain backward compatibility or remove this and treat it as a custom vendor extension if a specific implementation cannot pivot to the updated syntax immediately?

Reply:
I think the latter is simpler. Vendors can always accept an extended language.